packages/terra-date-input/src/DateInput.jsx

```diff
@@ -0,0 +1,644 @@
+      this.handleValueChange(event, DateInputUtil.inputType.YEAR, stateValue);
+    }
+
+    if (event.keyCode === KeyCode.KEY_LEFT) {
+      // If cursor at beginning of input
+      if (this.yearRef.selectionEnd === 0) {
+        if (displayFormat === 'month-day-year') {
+          this.focusDay(event);
+        }
+
+        if (displayFormat === 'day-month-year') {
+          this.focusMonth(event);
+        }
+      }
+    }
+
+    if (event.keyCode === KeyCode.KEY_BACK_SPACE || event.keyCode === KeyCode.KEY_DELETE) {
+      if (displayFormat === 'month-day-year' && event.target.value === '') {
+        this.focusDay(event);
+      }
+
+      if (displayFormat === 'day-month-year' && event.target.value === '') {
+        this.focusMonth(event);
+      }
+    }
+  }
+
+  handleValueChange(event, type, value) {
+    if (type === DateInputUtil.inputType.MONTH) {
+      this.setState({
+        month: value,
+        monthIsFocused: false,
+      });
+    } else if (type === DateInputUtil.inputType.DAY) {
+      this.setState({
+        day: value,
+        dayIsFocused: false,
+      });
+    } else if (type === DateInputUtil.inputType.YEAR) {
+      this.setState({
+        year: value,
+        yearIsFocused: false,
+      });
+    }
+
+    if (this.props.onChange) {
+      const month = type === DateInputUtil.inputType.MONTH ? value : this.state.month;
+      const day = type === DateInputUtil.inputType.DAY ? value : this.state.day;
+      const year = type === DateInputUtil.inputType.YEAR ? value : this.state.year;
+
+      if (month === '' && day === '' && year === '') {
+        this.handleOnChange(event, '');
+      } else {
+        const dateString = `${year}-${month}-${day}`;
+        this.handleOnChange(event, dateString);
+      }
+    }
+  }
+
+  /**
+   * Calls onChange callback prop if it exists
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {String} dateString Returns the dateString from the component
+   */
+  handleOnChange(event, dateString) {
+    if (this.props.onChange) {
+      this.props.onChange(event, dateString);
+    }
+  }
+
+  /**
+   * Renders month input
+   */
+  monthRender() {
+    return (
+      <div className={cx('date-input-month-wrapper')}>
+        <select
+          {...this.props.monthAttributes}
+          aria-label={this.props.intl.formatMessage({ id: 'Terra.date.input.monthLabel' })}
+          ref={this.monthRef}
+          className={cx('date-input-month', { 'is-focused': this.state.monthIsFocused })}
+          value={this.state.month}
+          name={'terra-date-month-'.concat(this.props.name)}
+          onChange={this.handleMonthChange}
+          onKeyDown={this.handleMonthInputKeyDown}
+          onFocus={this.handleMonthFocus}
+          onBlur={this.handleMonthBlur}
+          disabled={this.props.disabled}
+        >
+          <option value="" disabled>{this.props.intl.formatMessage({ id: 'Terra.date.input.monthPlaceholder' })}</option>
```

Comment:
Is it intentional that it only allows to use the Back/Delete keys to remove a selected month and not use the `allowClear` prop on Select to click on this option to clear it?

Reply:
This is using the native HTML select so the allowClear prop isn't available. There were some issues getting onKeyDown to work with the terra-select so I opted for a native-select.